test/test.js

```diff
@@ -36,4 +37,96 @@ describe('Database interactions', () => {
       id += 1;
     }
   });
+  test('Should get names for all product id entries', (done) => {
+    let id = 0;
+    const queryString = 'SELECT * FROM user_reviews WHERE url_id = ?';
+    const promises = [];
+    while (id < 100) {
+      expect.assertions(100);
+      db.query(queryString, [id], (err, results) => {
+        promises.push(results);
+        if (promises.length === 100) {
+          for (let i = 0; i < promises.length; i += 1) {
+            expect(promises[i][0].name).toBeDefined();
+          }
+          done();
+        }
+      });
+      id += 1;
+    }
+  });
+  test('Should get ratings for all product id entries & should be between 1-5', (done) => {
+    let productId = 0;
+    const queryString = 'SELECT * FROM user_reviews WHERE url_id = ?';
+    const promises = [];
+    while (productId < 100) {
+      expect.assertions(200);
```

Comment:
I'm curious. Why do you need the expect.assertions again? I would expect it to fail the first 99 times the while loop completes. I'm assuming it doesn't, but it seems to be unnecessary. If you do need it, wouldn't it be cleaner to have it outside the while loop? Or, you could put it inside the if(promises.length === 100) block.

Reply:
Great suggestion: inside the if statement makes much more sense as I can catch any errors up to that point. I'll make this change!